### ISSUE TITLE
Warn user when installing native dependency to `./package.json`

### ISSUE THREAD

**Issue by amilajack** [chentsulin/electron-react-boilerplate#1171]
Throw a warning message explaining that webpack does not work with native dependencies. Provide the link to further documentation. Provide directions on how to properly install the native dependency. This validation should run as a `postinstall` event.

```js
fs.readdirSync('node_modules')
  .filter(x => ['.bin'].indexOf(x) === -1)
```

Taken from http://jlongster.com/Backend-Apps-with-Webpack--Part-I

**Comment by uppe-r** [comment]
Hey, I might be able to make a PR about this today or tomorrow, but I would like to know if you agree with my vision @amilajack: 

- The module just added is checked on ```preinstall```. In case it is a native module, a colorful prompt appears informing the user that he should install it inside the app folder in the console, link to the wiki page, etc. The user will have the option to continue installing the module or cancel the install.

**Comment by amilajack** [comment]
Agree! I didn't even know there was a preinstall hook. Make sure to add the script in `./internals/scripts/CheckNativeDep.js`

**Comment by uppe-r** [comment]
Hey @amilajack I started looking into this yesterday, and I'd like to have your opinion on some issues:

The script you posted (taken from that link) does... nothing? It just checks for other folders rather than `.bin` which doesn't help with the issue as far as I'm aware. **So, the question is: how to identify a native dependency?**

From what I've searched, it looks like a `binding.gyp` file is an indicator of such, but not in some of the ones referenced as examples in the wiki (`node-postgres`, for example, which is just `JS `and `Makefile`, no any kind of `C`) so I guess some of those are there by mistake, right?

Also, my `preinstall` idea is kind of impossible, since the module files can't be checked at that time. I was thinking about still having the prompt and giving the user the choice to remove the module just added and install it on the proper folder automatically.

**Comment by amilajack** [comment]
The lead I gave you on this was pretty terrible. [Here's a better one](https://github.com/electron/electron-rebuild/blob/master/src/rebuild.ts#L34-L44):

```js
const locateNodeGyp = async () => {
  let testPath = __dirname;
  for (let upDir = 0; upDir <= 20; upDir++) {
    const nodeGypTestPath = path.resolve(testPath, `node_modules/.bin/node-gyp${process.platform === 'win32' ? '.cmd' : ''}`);
    if (await fs.exists(nodeGypTestPath)) {
      return nodeGypTestPath;
    }
    testPath = path.resolve(testPath, '..');
  }
  return null;
};
```

**Comment by aguynamedben** [comment]
This woulda saved me a bunch of time! Awesome 👍

**Comment by uppe-r** [comment]
Hey @amilajack, sorry to be bothering you again, **but how do you identify a native dependency analyzing the files**? I still don't understand that, since almost all if not all the modules described in the Wiki as native have no C/C++/C# code in them, and upon installing them there's no file created in `node_modules/.bin` which is what that piece code should be used to check, right?

**Comment by amilajack** [comment]
Sure, I'd be happy to explain. If `./node_modules/{dependency}/binding.gyp` exists then that dependency is a native dependency.

**Comment by uppe-r** [comment]
That's what I thought (as described in one of my previous comments)! Still, some of the native modules, if not all referenced in the Wiki have no `binding.gyp` file or use C/C++, that's why I was getting confused. I'll make the PR soon, had some things to take care of.
